flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -794,7 +828,7 @@ public void run() {
 
 		@Override
 		public void disconnected(SchedulerDriver driver) {
-			runAsync(new Runnable() {
+			runAsyncWithoutFencing(new Runnable() {
```

Comment:
@GJL do you still remember why this message needs to be run w/o fencing?

Reply:
I have a suspicion why we did it. If you have time for a quick call, then we could discuss it and maybe it rings a bell. Just let me know when it would work for you.

In any case, changing it and running it on `flink-jepsen` sounds like a good idea to validate the changes. cc @xintongsong.